lib/__tests__/_sendEvent.node.test.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * @jest-environment node
+ */
+import AlgoliaAnalytics from "../insights";
+import { getRequesterForNode } from "../utils/getRequesterForNode";
+import { getFunctionalInterface } from "../_getFunctionalInterface";
+import { setUserToken } from "../_cookieUtils";
+
+const credentials = {
+  apiKey: "testKey",
+  appId: "testId"
+};
+
+const defaultPayload = {
+  eventName: "my-event",
+  index: "my-index",
+  objectIDs: ["1"]
+};
+
+describe("_sendEvent in node env", () => {
+  let aa;
+  beforeEach(() => {
+    const instance = new AlgoliaAnalytics({ requestFn: getRequesterForNode() });
+    aa = getFunctionalInterface(instance);
+    aa("init", credentials);
+  });
+
+  it("throws when user token is not set", () => {
+    expect(() => {
+      aa("sendEvent", "click", {
+        ...defaultPayload
+      });
+    }).toThrowError(
+      "Before calling any methods on the analytics, you first need to call 'setUserToken' function or include 'userToken' in the event payload."
+    );
+  });
+
+  it("does not throw when user token is set", () => {
+    aa("setUserToken", "aaa");
+    expect(() => {
+      aa("sendEvent", "click", {
+        ...defaultPayload
+      });
+    }).not.toThrowError();
+  });
+
+  it("does not throw when user token is set inside payload", () => {
+    expect(() => {
+      aa("sendEvent", "click", {
+        userToken: "aaa",
+        ...defaultPayload
+      });
+    }).not.toThrowError();
+  });
+
+  it("throws when user token in payload is not a string", () => {
+    expect(() => {
+      aa("sendEvent", "click", {
+        userToken: 3,
+        ...defaultPayload
+      });
+    }).toThrowError("expected optional parameter `userToken` to be a string");
+  });
+
+  it("throws when user token is an empty string", () => {
+    expect(() => {
+      aa("setUserToken", "");
+      aa("sendEvent", "click", {
+        ...defaultPayload
+      });
+    }).toThrowError("`userToken` cannot be an empty string.");
+
+    expect(() => {
+      aa("sendEvent", "click", {
+        userToken: "",
+        ...defaultPayload
+      });
+    }).toThrowError("`userToken` cannot be an empty string.");
```

Comment:
To be 100% rigorous I would split this into to different `it` to fully reset the state between the first call and second call in the beforeEach.

Reply:
Oh right. I didn't think of resetting the state between cases. https://github.com/algolia/search-insights.js/pull/140/commits/ad4bea2995860435be516a68b098da7b7121ad42